pkg/languagedetection/cli_detector.go

```diff
@@ -0,0 +1,75 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+package languagedetection
+
+import (
+	"strings"
+
+	"github.com/DataDog/datadog-agent/pkg/languagedetection/languagemodels"
+	"github.com/DataDog/datadog-agent/pkg/process/procutil"
+)
+
+type languageFromCLI struct {
+	name      languagemodels.LanguageName
+	validator func(exe string) bool
+}
+
+// knownPrefixes maps languages names to their prefix
+var knownPrefixes = map[string]languageFromCLI{
+	"python": {name: languagemodels.Python},
+	"java": {name: languagemodels.Java, validator: func(exe string) bool {
+		if exe == "javac" {
+			return false
+		}
+		return true
```

Comment:
Not sure whether this validator is going to be extended in the future, but if there are no plans to make it more complicated, it seems like this could be rewritten as `return exe != "javac"`

Reply:
This isn't actually new code, sorry, the git blame got messed up because I renamed the file. I ended up making it like this to extend the validator. I'd prefer to avoid touching this code as much as possible, but if you feel it's a blocker I can change it.